src/Api/FilterInterface.php

```diff
@@ -36,4 +36,13 @@
      * @return array
      */
     public function getDescription(string $resourceClass): array;
+
+    /**
+     * setter for properties.
+     *
+     * used in the dependency injection to allow independent constructor arguments.
+     *
+     * @param array $properties
+     */
+    public function setProperties(array $properties);
```

Comment:
Whoops, breaking change. Any ideas on how I could avoid this? I've added this because I create services on the fly in the Dependency injection, and because the constructor arguments aren't predictable I can't do `$def->replaceArgument`.